lib/kitchen/driver/azurerm.rb

```diff
@@ -354,7 +366,7 @@ def validate_state(state = {})
         state[:uuid] = SecureRandom.hex(8) unless existing_state_value?(state, :uuid)
         state[:server_id] = "vm#{state[:uuid]}" unless existing_state_value?(state, :server_id)
         state[:azure_resource_group_name] = azure_resource_group_name unless existing_state_value?(state, :azure_resource_group_name)
-        %i{subscription_id vm_name azure_environment use_managed_disks}.each do |config_element|
+        %i{subscription_id username password vm_name azure_environment use_managed_disks}.each do |config_element|
```

Comment:
Re-adding these will reintroduce #118, since an invalid password will once again be persisted to the state file even if the deployment fails.

Reply:
Per our slack conversation:

```text
kitchen destroy will in fact succeed even when the statefile has an invalid password
```

Reply:
Applied patch and tested:

* default password generation works
* setting `password` in kitchen.yml or kitchen.local.yml works in the `[driver]` section
* using an environment variable works